### PR TITLE
feat: add builder check syntax plugin

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Description

新增不兼容语法检测插件

1. 成功时
image

2. 失败时
img_v2_2568a6de-a970-4567-ae24-3d30b661adeg

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-doc/docs/zh/config/security/checkSyntax.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+```ts
+export default {
+  security: {
+    checkSyntax: true,
+  },
+};
+```
+
+如果将 `security.checkSyntax` 指定为 `true` 的话，则 targets 会被认定为项目设定的 browserslist。详情请参考 [设置浏览器范围](http://edenx.bytedance.net/builder/zh/guide/advanced/browserslist.html)
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/zh/config/security/checkSyntax.md:23]: 换成外网域名

File: packages/builder/builder-doc/docs/en/config/security/checkSyntax.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export default defineConfig({
+  plugins: [appTools()],
+  security: {
+    checkSyntax: true,
+  },
+});
+```
+
+If `security.checkSyntax` is specified as `true`, targets will be recognized as the browserslist set by the project. For details, please refer to [Browserslist](http://edenx.bytedance.net/builder/en/guide/advanced/browserslist.html)
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/en/config/security/checkSyntax.md:26]: 换成外网域名

File: packages/builder/builder-doc/docs/en/config/security/checkSyntax.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+```
+
+- **Default**: `false`
+
+Analyzes the generated product for the presence of high-level syntax that is not compatible in the specified environment. If present, the details are printed in the terminal.
+
+## Example
+
+```ts title="edenx.config.ts"
```
Comment (chenjiahan) [inline packages/builder/builder-doc/docs/en/config/security/checkSyntax.md:15]: edenx -> modern-js

File: packages/builder/builder-doc/docs/zh/config/security/checkSyntax.md (hunk 1 above)
Comment (GiveMe-A-Name) [reply packages/builder/builder-doc/docs/zh/config/security/checkSyntax.md]: done

File: packages/builder/builder-doc/docs/en/config/security/checkSyntax.md (hunk 2 above)
Comment (GiveMe-A-Name) [reply packages/builder/builder-doc/docs/en/config/security/checkSyntax.md]: done

File: packages/builder/builder-doc/docs/en/config/security/checkSyntax.md (hunk 3 above)
Comment (GiveMe-A-Name) [reply packages/builder/builder-doc/docs/en/config/security/checkSyntax.md]: It should support more framework for builder, it should not bind with modern.js.

so delete the `titel=edenx(modern).config.ts`

**Review by chenjiahan — APPROVED** [review]

**Comment by fireairforce** [comment]
awesome

**Pull request merged**